### PR TITLE
added documentation for archiver.py and created an rst file for sphinx

### PR DESCRIPTION
**Pull request by jpgtordilla**

### CODE REVIEW

File: lcls_tools/common/data_analysis/archiver.py [hunk 1]
```diff
@@ -44,6 +56,14 @@ def __hash__(self):
 
     @property
     def timestamp(self) -> datetime:
+        """Gets the datetime object for the ArchiverValue data point at the given time.
+
+        matplotlib can handle datetime objects directly, but data should be cleaned so that HH:MM:SS are also displayed.
+
+        :return self._timestamp: a datetime object corresponding to a data point in the dataset.
+        :rtype self._timestamp: datetime
```
Comment (lisazacarias) [inline lcls_tools/common/data_analysis/archiver.py:64]: Question: is there something parsing this that requires having the type annotation twice? As in, the `-> datetime` is already there

Comment (jpgtordilla) [reply lcls_tools/common/data_analysis/archiver.py:64]: No, I think I was going a bit overboard with the docstring annotations. The `-> datetime` is enough.

Comment (lisazacarias) [reply lcls_tools/common/data_analysis/archiver.py:64]: Oh ok! I think we can probably get rid of all the excess annotations then. All of the function signatures should have type annotations (if they don't, feel free to add them)

Comment (jpgtordilla) [reply lcls_tools/common/data_analysis/archiver.py:64]: Ok will do.

**Review by lisazacarias — APPROVED** [review]
Looks great! Thanks for doing this!

**Pull request closed without merging**